arangod/VocBase/VocBaseLogManager.cpp

```diff
@@ -121,46 +121,44 @@ auto VocBaseLogManager::updateReplicatedState(
   }
 }
 
+void VocBaseLogManager::prepareDropAll() noexcept {
+  auto guard = _guardedData.getLockedGuard();
+  guard->resignAllWasCalled = true;
```

Comment:
do we need an (if (guard->resignAllWasCalled == true) { return; }) 
Before this line?
To protect against 2 Threads triggering DropAll() ?
Then one will wait on the other, but both will actually excecute below calls.

Reply:
Although it wouldn't hurt, that's not necessary. A second `prepareDropAll()` would just find and empty `statesAndLogs` map and basically do nothing.